### DISCUSSION TITLE
Pre-Release Versions

### DISCUSSION THREAD

**Discussion by alienatedsec** [alienatedsec/yi-hack-v5#248] [Show and tell]
`[Updated] 2024/03/01`

****The below stopped working with baseline `0.3.8` and was fixed with a new baseline [0.4.1](https://github.com/alienatedsec/yi-hack-v5/releases/tag/0.4.1) - more details #326****

Dear All

Unfortunately, my time is limited, and I want this project to carry on. Therefore, I am trying something new, and I hope it will be popular to an extent.

I am introducing pre-releases, and I need the community to test those and let me know if there are any issues as part of this discussion. Issues for a pre-release should not be raised, but discussed over here. Some guidance about the pre-release firmware versions based on the example - `0.4.1j`

`0` - major version release
`4` - the main version that is going to be released
`1j` - the minor pre-release version

The above pre-release once tested and approved, will become the latest release e.g., `0.4.1`

Pre-releases will be available via [official releases](https://github.com/alienatedsec/yi-hack-v5/releases) and will require manual or WebUI-supported installation. The latter is preferred due to many conditions.

## [Preferred] WebUI Support for Pre-Releases started with version `0.3.8` and now it needs `0.4.1` to work properly.

![image](https://github.com/alienatedsec/yi-hack-v5/assets/73167064/06af91db-40c8-46a8-a758-1dfa55fb48c6)

# Some simplified manual steps are below:
- Copy the respective firmware file to `/tmp/sd/` e.g., `yi_home_1080p_0.4.1j.tgz`
- Change its name to `yi_home_1080p_x.x.x.tgz` - yes, it has to be `x` and NOT any random number or letter.
- You can press `UPGRADE to LATEST` or `UPGRADE to PRE-RELEASE` buttons
- The process will pick the file in the `/tmp/sd/` and upgrade your camera to that pre-release version.

You can revert from any pre-release to the latest official release by ignoring the above steps, going to the `Service` page and simply pressing `UPGRADE to LATEST`.

Looking forward to your support and helping to improve this project. Thank you

**Comment by Loafdude** [comment]
Firstly, thank you for supporting this project. It is very much appreciated!

I am testing with a Yi home 1080p. I have outdoor units avaialble to test with as well.
I have tested your 9.3.8 pre-release with a single high-res stream w/audio enabled.
Audio works for a few moments then becomes studders then completely fails.
Video is studdery too but I believe this is a side effect of the audio.

Video only stream works fine. Audio only stream works fine

Client is VLC 3.0.18
Here are the stats I get from VLC
![image](https://user-images.githubusercontent.com/17401709/232380450-4f803ad5-030d-4a8e-b547-82175bd8dbd3.png)

Please let me know if I can assist further.

**Reply by alienatedsec** [reply]
Thanks @Loafdude - it feels like this `9.3.8` pre-release has several issues already:
- RTSP is not good when both video and audio streaming
- MQTT needs re-working - testing another branch
- I am not sure if you experienced Snapshot issues (?)

For those simple reasons, I recommend waiting for the `9.3.8a`.

**Reply by Loafdude** [reply]
Snapshots seem to be working here. 
Any specific issue I should test with it?

I did not test MQTT, sorry.

**Comment by roleoroleo** [comment]
@alienatedsec
We should go into detail about rtsp when both video and audio are enabled.
It could be a cpu/memory issue or it could be a sync problem between the streams (maybe with the timestamps of the frames).

**Reply by alienatedsec** [reply]
@roleoroleo The CPU and Memory are not a problem as hardly utilised - but if you have any ideas for sync, that could be a way to resolve it.

**Reply by roleoroleo** [reply]
My doubt is that the presentation time of the video frame is different from the presentation time of the audio frame.
But I checked the code and it seems to be ok.
I will check again it with tcpdump.

**Reply by alienatedsec** [reply]
@roleoroleo check this out as looks very similar https://github.com/AlexxIT/go2rtc/issues/338

**Reply by roleoroleo** [reply]
I will take a look.
...
I don't know it's a problem but when I close the webrtc window, the rtsp stream between ha and the cam doesn't stop.
Maybe the problem is when the stream is restarted?
I have to deepen.

**Reply by alienatedsec** [reply]
@roleoroleo My latest commit eb40cb4 combined with yours 31b7bcb seems to be fixing the MQTT and other things I didn't even realise were wrong; however, it does not fix the audio & video combined performance. The below from VLC could be also relevant

![image](https://user-images.githubusercontent.com/73167064/233221199-6d1804e2-708d-4277-bd60-52b14222a225.png)

**Reply by roleoroleo** [reply]
Was the stream stable with the previous version?

**Reply by alienatedsec** [reply]
I think its the same; however, it is stable without audio even on the latest version @roleoroleo

**Reply by roleoroleo** [reply]
Is there any difference if you configure RTSP resolution both, high or low?
I mean with audio on.

Could you send me a capture of the stream when it doesn't work?

**Reply by alienatedsec** [reply]
`/tmp/view` will produce the same regardless of the settings (audio, no audio), so what is the best you would like to get? Any commands to capture? @roleoroleo

**Reply by roleoroleo** [reply]
I mean the rtsp stream captured with tcpdump/wireshark.
I would like to check the timestamps inside the rtp packets.

**Reply by Loafdude** [reply]
Here is a capture of my bad audio/video combo with wireshark
https://drive.google.com/file/d/1Mqb2pFiZoHwd_BMchG8pVuCqABRNrFtK/view?usp=sharing

**Reply by alienatedsec** [reply]
Thanks @Loafdude I am about to share the same shortly

**Reply by roleoroleo** [reply]
I noticed a problem with the sample rate.
Are you able to capture 1 minute of the raw streams (both audio and video)?
And share the files with me?
You can do it stopping rRTSPServer and reading the fifo files
/tmp/aac_audio_fifo
/tmp/h264_high_fifo

**Reply by alienatedsec** [reply]
> /tmp/aac_audio_fifo
> /tmp/h264_high_fifo

@roleoroleo shared already via email

**Reply by roleoroleo** [reply]
Do you want to try this patch?
[p.patch.gz](https://github.com/alienatedsec/yi-hack-v5/files/11289968/p.patch.gz)

**Reply by Loafdude** [reply]
I compiled sucessfully with this patch but it did not help my audio/video issues. 
That said, my dev environment is very janky. I'd really like it if @alienatedsec could compile to confirm.

**Comment by alienatedsec** [comment]
`9.3.8a` is now ready to test - still, sync problems when both `audio and video` stream together, but it works fine without audio. The rest needs testing - It is important to flash it with FAT32 SD Card as some libraries cannot be loaded from the SD card and had to be copied to the `/home/yi-hack-v5/lib` folder. The home partition, once flashed, should already have all the necessary files, but `system.sh` will copy those libraries if they don't exist. Regardless, there are changes to `system_init.sh` which also loads from the `/home/yi-hack-v5/scripts/` folder and will need to be correct to avoid double-loading of certain files. You will see `fat32error.txt` in your `/tmp/sd/` if flashing wasn't fully performed.

**Reply by roleoroleo** [reply]
If the new ipc multiplexer is a problem I could study a new solution (different from MStar and Allwinner) trying to avoid the use of LD_PRELOAD.

**Reply by alienatedsec** [reply]
the LD_PRELOAD works  in the latest design :) it helped me identify other issues with the config file :) - thanks @roleoroleo 
Any ideas about how to fix sync issues? I added a comment with a screenshot from the VLC.

**Comment by jidehem92** [comment]
Hi. Thank you for your work. I've tested 9.3.8a with my yi_outdoor and I confirm it fixes the version problem in Home Assistant. I don't have any problem yet with this version (I don't stream audio)

**Comment by oat1** [comment]
@alienatedsec 
I've installed the latest 9.3.8b version

I have a problem with mqtt, most of sensors are unavailable.

![Screenshot_20230423_215201_Home Assistant](https://user-images.githubusercontent.com/59432560/233862201-4f5aaf20-b59b-459c-ad45-23bc0906c909.jpg)

**Reply by alienatedsec** [reply]
@oat1 are you on the latest ha integration?

Also, do you have the error file on your sd card in `/tmp/sd/` ?

**Comment by oat1** [comment]
I do not see any errors
![Screenshot_20230423_222017_RS File Manager](https://user-images.githubusercontent.com/59432560/233863877-8f6f7734-2537-4a2e-b9d4-f13268c4fe89.jpg)

I'm on 0.4.5 yihack integration

**Reply by alienatedsec** [reply]
```
killall mqttv4
mqttv4
```

Can you share the output after running the above commands?

**Reply by alienatedsec** [reply]
sometimes reloading the integration also helps

**Reply by oat1** [reply]
@alienatedsec 
how to login via ssh and run killall mqttv4 ?
when I try, I get 
<img width="678" alt="image" src="https://user-images.githubusercontent.com/59432560/233909405-cb66d583-7b4f-47ae-8fcd-d7dc8af1a676.png">

I also tried 
<img width="455" alt="image" src="https://user-images.githubusercontent.com/59432560/233957616-dd63bec4-935b-4e1e-bdca-be053ec7f0bd.png">

I'm on HassOS using addon SSH & Web Terminal 13.1.0

And by the way, I do not see sensors for these toggles :

<img width="250" alt="image" src="https://user-images.githubusercontent.com/59432560/233909597-4aa2e175-f818-47cd-9eed-0b9492394b1a.png">

<img width="253" alt="image" src="https://user-images.githubusercontent.com/59432560/233909625-85de468a-b06b-47fb-8244-78640bf75f5a.png">

I got this error 
`Logger: homeassistant.components.binary_sensor
Source: custom_components/yi_hack/binary_sensor.py:96
Integration: Sensor binarny (documentation, issues)
First occurred: 09:33:30 (1 occurrences)
Last logged: 09:33:30

Error while setting up yi_hack platform for binary_sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 304, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/yi_hack/binary_sensor.py", line 41, in async_setup_entry
    YiMQTTBinarySensor(config, "human_detection"),
  File "/config/custom_components/yi_hack/binary_sensor.py", line 96, in __init__
    self._payload_on = config.data[CONF_HUMAN_DETECTION_MSG]
KeyError: 'AI_HUMAN_DETECTION_MSG'
`

**Reply by roleoroleo** [reply]
It seems that your mqtt.conf doesn't contain AI_HUMAN_DETECTION_MSG line.
Could you check please?

**Reply by oat1** [reply]
@roleoroleo 
Where can I find mqtt.conf ?

**Reply by oat1** [reply]
@alienatedsec I wasn't able to connect via ssh but I was able to connect via telnet . I did killall mqttv4 and then mqttv4 command, and i got this
```
Connecting to 192.168.2.110 port 23, please wait...





      _     _           _           ___ 
  _ _|_|___| |_ ___ ___| |_ ___ _ _|  _|
 | | | |___|   | .'|  _| '_|___| | |__ |
 |_  |_|   |_|_|__,|___|_,_|    \_/|___|
 |___|
 -----------------------------------------------------
  yi-hack-v5 - 9.3.8b
 -----------------------------------------------------
 WARNING: Cannot check for updates from GitHub.

 The camera might not be connected to the internet
 or, if it has just been turned on, the updater
 hasn't contacted the server yet.
 -----------------------------------------------------

/home/yi-hack-v5 # killall mqttv4
/home/yi-hack-v5 # mqttv4
Starting mqttv4 v0.1.0
key: HTTPD | value: yes
Unrecognized config line, ignore it
key: TELNETD | value: yes
Unrecognized config line, ignore it
key: SSHD | value: yes
Unrecognized config line, ignore it
key: FTPD | value: yes
Unrecognized config line, ignore it
key: BUSYBOX_FTPD | value: no
Unrecognized config line, ignore it
key: DISABLE_CLOUD | value: no
Unrecognized config line, ignore it
key: REC_WITHOUT_CLOUD | value: yes
Unrecognized config line, ignore it
key: RTSP | value: yes
Unrecognized config line, ignore it
key: RTSP_STREAM | value: low
Unrecognized config line, ignore it
key: RTSP_AUDIO | value: no
Unrecognized config line, ignore it
key: ONVIF | value: no
Unrecognized config line, ignore it
key: ONVIF_WSDD | value: no
Unrecognized config line, ignore it
key: ONVIF_PROFILE | value: low
Unrecognized config line, ignore it
key: ONVIF_NETIF | value: wlan0
Unrecognized config line, ignore it
key: ONVIF_WM_SNAPSHOT | value: no
Unrecognized config line, ignore it
key: NTPD | value: yes
Unrecognized config line, ignore it
key: NTP_SERVER | value: pool.ntp.org
Unrecognized config line, ignore it
key: PROXYCHAINSNG | value: no
Unrecognized config line, ignore it
key: SWAP_FILE | value: yes
Unrecognized config line, ignore it
key: RTSP_PORT | value: 554
Unrecognized config line, ignore it
key: ONVIF_PORT | value: 80
Unrecognized config line, ignore it
key: HTTPD_PORT | value: 8080
Unrecognized config line, ignore it
key: TIMEZONE | value: CET-1CEST,M3.5.0,M10.5.0/3
Unrecognized config line, ignore it
key: FREE_SPACE | value: 5
Unrecognized config line, ignore it
key: FTP_UPLOAD | value: no
Unrecognized config line, ignore it
key: FTP_DIR_TREE | value: no
Unrecognized config line, ignore it
key: FTP_FILE_DELETE_AFTER_UPLOAD | value: yes
Unrecognized config line, ignore it
key: CHECK_UPDATES | value: no
Unrecognized config line, ignore it
key: TOPIC_AI_HUMAN_DETECTION | value: ai_human_detection
Unrecognized config line, ignore it
key: TOPIC_BABY_CRYING | value: baby_crying
Unrecognized config line, ignore it
key: AI_HUMAN_DETECTION_START_MSG | value: human_start
Unrecognized config line, ignore it
key: AI_HUMAN_DETECTION_STOP_MSG | value: human_stop
Unrecognized config line, ignore it
key: SOUND_DETECTION | value: sound
Unrecognized config line, ignore it
key: MQTT_RETAIN_AI_HUMAN_DETECTION | value: 1
Unrecognized config line, ignore it
key: MQTT_RETAIN_BABY_CRYING | value: 1
Unrecognized config line, ignore it
Trying to connect... Client KameraTaras sending CONNECT
Client KameraTaras received CONNACK (0)
Client KameraTaras sending PUBLISH (d0, q1, r1, m1, 'yicam2/status', ... (6 bytes))
.
connected!
Client KameraTaras received PUBACK (Mid: 1)
Client KameraTaras sending PINGREQ
Client KameraTaras received PINGRESP

``` 

It changed nothing in home assistant

**Reply by alienatedsec** [reply]
Can you access /tmp/sd/ from telnet?

**Reply by oat1** [reply]
I can access it from ftp
![Screenshot_20230427_221035_RS File Manager](https://user-images.githubusercontent.com/59432560/234979746-3c53229b-6c57-47f5-9620-422c16156dbe.jpg)

**Reply by roleoroleo** [reply]
> @roleoroleo Where can I find mqtt.conf ?

/tmp/sd/yi-hack-v5/etc/mqttv4.conf

**Reply by oat1** [reply]
@alienatedsec , @roleoroleo 
Any chances for fix to Mqtt ?
Right now with ver 9.3.8b, all sensors are down and all my automations with movement detection don't work anymore :(

**Reply by roleoroleo** [reply]
Please post your mqttv4.conf
Login with telnet and read the file /tmp/sd/yi-hack-v5/etc/mqttv4.conf

**Reply by oat1** [reply]
@roleoroleo 
```
###############################################################################
#                       Configuration file for mqttv4                         #
###############################################################################

# -----------------------------------------------------------------------------
# Basic MQTT settings
# -----------------------------------------------------------------------------

MQTT_IP=192.168.2.115
MQTT_PORT=1883

MQTT_CLIENT_ID=KameraTaras

MQTT_USER=mqtt
MQTT_PASSWORD=Password

MQTT_PREFIX=yicam2
MQTT_PREFIX_CONFIG=

# -----------------------------------------------------------------------------
# Set the topics where the messages will be published
# The final topic will be MQTT_PREFIX/TOPIC_MOTION
# -----------------------------------------------------------------------------

TOPIC_BIRTH_WILL=status
TOPIC_AI_HUMAN_DETECTION=ai_human_detection
TOPIC_MOTION=motion_detection
TOPIC_MOTION_IMAGE=motion_detection_image
TOPIC_MOTION_FILES=motion_files
TOPIC_BABY_CRYING=baby_crying
TOPIC_SOUND_DETECTION=sound_detection

# -----------------------------------------------------------------------------
# Set the topics messages
# -----------------------------------------------------------------------------

BIRTH_MSG=online
WILL_MSG=offline
AI_HUMAN_DETECTION_START_MSG=human_start
AI_HUMAN_DETECTION_STOP_MSG=human_stop
MOTION_START_MSG=motion_start
MOTION_STOP_MSG=motion_stop
BABY_CRYING_MSG=crying
SOUND_DETECTION=sound

# -----------------------------------------------------------------------------
# Other settings
# -----------------------------------------------------------------------------

MQTT_KEEPALIVE=120
MQTT_QOS=1
MQTT_RETAIN_BIRTH_WILL=1
MQTT_RETAIN_AI_HUMAN_DETECTION=1
MQTT_RETAIN_MOTION=1
MQTT_RETAIN_MOTION_IMAGE=1
MQTT_RETAIN_MOTION_FILES=1
MQTT_RETAIN_BABY_CRYING=1
MQTT_RETAIN_SOUND_DETECTION=1
SOUND_DETECTION_MSG=sound_detected
AI_HUMAN_DETECTION_MSG=human
```

**Reply by roleoroleo** [reply]
AI_HUMAN_DETECTION_MSG is present.
Is the error the same as above now?
```
Logger: homeassistant.components.binary_sensor
Source: custom_components/yi_hack/binary_sensor.py:96
Integration: Sensor binarny (documentation, issues)
First occurred: 09:33:30 (1 occurrences)
Last logged: 09:33:30

Error while setting up yi_hack platform for binary_sensor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 304, in _async_setup_platform
await asyncio.shield(task)
File "/config/custom_components/yi_hack/binary_sensor.py", line 41, in async_setup_entry
YiMQTTBinarySensor(config, "human_detection"),
File "/config/custom_components/yi_hack/binary_sensor.py", line 96, in init
self._payload_on = config.data[CONF_HUMAN_DETECTION_MSG]
KeyError: 'AI_HUMAN_DETECTION_MSG'
```

**Reply by oat1** [reply]
No, I do not see this error anymore, it must be a one-time.
I do not see any additional errors in HA Logs. But all sensors are unavailable.

I do not know if this is related but the camera also very often becomes unavailable

<img width="230" alt="image" src="https://github.com/alienatedsec/yi-hack-v5/assets/59432560/8b61723f-4b75-4ce9-a2ed-1f3c4aa95d4b">

**Reply by roleoroleo** [reply]
Probably the cam is in boot loop.
You have to wait a new release from alienatedsec.

**Reply by alienatedsec** [reply]
> I do not know if this is related but the camera also very often becomes unavailable

@oat1 Do you have sound enabled too? Try disabling sound, if so.

**Reply by oat1** [reply]
No I don't
![Screenshot_20230516_222040_Chrome](https://github.com/alienatedsec/yi-hack-v5/assets/59432560/84625531-b541-497f-ae98-f157ab38906d)

**Reply by oat1** [reply]
I've installed the latest 9.3.8.c
And the problem with unavailable sensors still appears 
![Screenshot_20230624_130827_Home Assistant](https://github.com/alienatedsec/yi-hack-v5/assets/59432560/84d4436a-85a7-4821-abfe-f201ac796432)

**Reply by alienatedsec** [reply]
@oat1 try turning those sensor-related switches on and off in the integration dashboard for your camera

**Reply by oat1** [reply]
I did, nothing changed. 
Then I deleted the cam from the HA and added it again. 

Now there are only 2 sensors :status and motion detection screen shot.

All the AI and motion detection are missing

![Screenshot_20230625_092253_Home Assistant](https://github.com/alienatedsec/yi-hack-v5/assets/59432560/709c83c7-9e5e-4a07-bdf7-1c39b10d3d63)

**Reply by roleoroleo** [reply]
Are you using the last version of the integration?

**Reply by oat1** [reply]
I'm using version 0.4.5, I believe this is the latest ?

**Reply by roleoroleo** [reply]
Yes, it's the last version.
@alienatedsec Is the tag 9.3.8c updated with the last changes?

**Reply by alienatedsec** [reply]
@roleoroleo yes it is. It's this branch https://github.com/alienatedsec/yi-hack-v5/tree/roleo_ipc

**Reply by roleoroleo** [reply]
I checked the code in the whole chain integration-hack and I didn't find problems.
But I can't test it with a cam.
@alienatedsec I commited a couple of updates but nothing important about the issue above.
Are you able to make a test with the integration?

**Comment by ghost** [comment]
Hello,

I tested several versions (camera is 27US) :
- yi_home_9.3.8 -> Snapshot works, stream in VLC too but integration in HA is broken (no video preview, the rest seems OK).
- yi_home_9.3.8a -> Unable to install (tried upgrade path through the interface and unpacking directly on fresh SD), the LED start blinking few seconds then go to solid orange.
-  yi_home_9.3.8b -> Same as yi_home_9.3.8a

Could you please check and provide me some help / fix ?
KR,

**Reply by Minims** [reply]
Hello,

>yi_home_9.3.8a -> Unable to install (tried upgrade path through the interface and unpacking directly on fresh SD), the LED start >blinking few seconds then go to solid orange.

I have the exact same issue :

**Reply by alienatedsec** [reply]
I found the problem with the size of the home partition being too big. I already have the resolution. There will be a new pre-release soon.

**Comment by b0b33140** [comment]
Hello,

I tried version 9.3.8b on my Yi Outdoor 9CUS camera and almost all of them work, I only have the following problems:
I can't save the videos because my sd card is formatted in FAT32. I guess if I format my SD card with the Yi home app, I'll switch to extfat.
Compatibility is partial with on home assistant with yi-hack Home Assistant integration, but I was able to view the RSTP stream otherwise.

Thanks for your work

**Reply by alienatedsec** [reply]
Hi @b0b33140 That's great to hear it works on your camera.

I currently get mixed messages and will need to decide when to release the next version. I will do some final testing and release it anyway.

Thanks a lot.

**Reply by b0b33140** [reply]
Hello, is there a way to save the videos on the sd card with the Pre-Release version?

**Reply by alienatedsec** [reply]
> Hello, is there a way to save the videos on the sd card with the Pre-Release version?

Is the SD formatted by the YI app?

**Reply by b0b33140** [reply]
No, you wrote: "It is important to flash it with FAT32 SD Card as some libraries cannot be loaded from the SD card and had to be copied to the /home/yi-hack-v5/lib folder."

**Reply by alienatedsec** [reply]
True. That is only for the initial hack and it's firmware loading. 

If you formatted as FAT32, and loaded the latest firmware you need to format it again with YI app and copy the yi-hack-v5 folder again, then restart. It's possible via FTP as it's enabled after restart by default when the SD is formatted by the app.

**Reply by alienatedsec** [reply]
@b0b33140

`fdisk -l` - `ID 83 Linux` allows recording on SD card (at least that is for my 8GB testing card), but that is only when formatted with the YI app.

![image](https://github.com/alienatedsec/yi-hack-v5/assets/73167064/5232115d-084d-4590-b69d-e0126b354f52)

**Reply by b0b33140** [reply]
Great, it works! On the other hand, my camera crashed, I had to take out the sd card from the camera and plug it into my pc to copy the "yi-hack-v5" folder.
Thank you very much

**Reply by alienatedsec** [reply]
`9.3.8d` seems to have a problem if there is no SD card or when it is formatted. I need to check what has been broken this time.

**Reply by alienatedsec** [reply]
@b0b33140 found it - https://github.com/alienatedsec/yi-hack-v5/blob/2df5c4dbbbe1bc6a5b296554a52ea6b8e471d23e/src/static/static/home/yi-hack-v5/script/system_init.sh#L47

It doesn't execute WiFi if no SD card 👎 I need to fix it with `9.3.8e`

**Comment by Jakdaw** [comment]
Hi!
Firstly, thanks for this project!

I've been using 0.3.7 on a Yi Outdoor cam for a little while now - in 'private' mode, sending RTSP to a Frigate instance (which has improved considerably with more recent versions of Frigate).

I tried to upgrade to 9.3.8c today (I'd love audio too!), but without much success:-

Following the upgrade process, the camera never came back onto the network. Just sat there with a Yellow LED.

Initially I thought this might be a physical SD card problem, so I switched to another card and went back to 0.3.7.

However, once I'd configured 0.3.7 again (& rebooted, all good). I tried again with the upgrade and got exactly the same outcome.

So I thought I'd wipe the SD card and try installing 9.3.8c in the same way I'd done 0.3.7 (ie root/home images and copy the yi-hack-v5 directory onto otherwise empty SD card). This worked! I did enable RTSP etc (no audio yet) but found that even low-res snapshots were triggering OOM killer, so I enabled swap. Rebooted and that was fine. Finally I thought I'd disable cloud again, since I don't use that - saved settings and reboot - and again, it never returns to the wifi network. Is there some problem with disabling Cloud for 9.3.8c? Please let me know what I can do to help diagnose this.

For now, I'm back on 0.3.7 exactly as things were from the start.

**Reply by alienatedsec** [reply]
cloudAPI management was rewritten here https://github.com/alienatedsec/yi-hack-v5/commit/7aa95ab9dd47c9a4a48e2ce6641a8ce75d775a8d

@roleoroleo could expand on this

**Reply by roleoroleo** [reply]
In this new cloudAPI management, the process is not removed but changed with a script that replies to the requests of the cloud (only some requests).
This approach fixes a lot of other problems.
My problem is always the same: I cannot test it to a v5 cam.

@alienatedsec Probably I missed some line for some models. I will try to update the scripts tosday.

**Reply by roleoroleo** [reply]
@alienatedsec I pushed a commit with some changes in cloudAPI.
https://github.com/alienatedsec/yi-hack-v5/commit/b7b55a8a04f7dbba16fffa7983c14ccdab848392
Could you check if the path are ok? I don't know the folder structure of the different v5 models.

**Reply by roleoroleo** [reply]
@alienatedsec Another question related to the problem above reported by Jakdaw).
The system.sh file starts yi daemon only if DISABLE_CLOUD == "no" or if REC_WITHOUT_CLOUD == "yes".
I don't know with v5 but with MStar and Allwinner, if dispatch and rmm are not running the cam doesn't start.
Dispatch probably controls the wifi and rmm starts the video processing.
What do you think about?

**Reply by alienatedsec** [reply]
@roleoroleo 
> The system.sh file starts yi daemon only if DISABLE_CLOUD == "no" or if REC_WITHOUT_CLOUD == "yes".

I have found something to potentially amend in here  https://github.com/alienatedsec/yi-hack-v5/blob/f0d34254cef32d9b4213eb1f2aba7f606a1d22cb/src/static/static/home/yi-hack-v5/script/cloudAPI#L24-L25 with the below so those are also executed

```
if [[ $(get_config DISABLE_CLOUD) == "yes" ]] ; then
        cd /home/app
        ./cloudAPI_fake "$@"
        killall dispatch
        LD_PRELOAD=/home/yi-hack-v5/lib/ipc_multiplex.so ./dispatch &
        sleep 2
        LD_LIBRARY_PATH="/home/yi-hack-v5/lib:/lib:/home/lib:/home/app/locallib:/home/hisiko/hisilib" ./rmm &
        sleep 4
```

I presume that `cloudAPI` is executed by `cloud` - do we need to keep the cloud running regardless? 
https://github.com/alienatedsec/yi-hack-v5/blob/f0d34254cef32d9b4213eb1f2aba7f606a1d22cb/src/static/static/home/yi-hack-v5/script/system.sh#L173-L174

**Reply by alienatedsec** [reply]
I just realised it will use the same command each time the `cloud` is calling `cloudAPI` - I need to add another line to start `./rmm` and `./dispatch` in `system.sh` whenever `if [[ $(get_config DISABLE_CLOUD) == "yes" ]] ; then`

**Reply by alienatedsec** [reply]
Also found this https://github.com/alienatedsec/yi-hack-v5/blob/e8701f2202c7c9f5d2c4b82dc3ccbf28276a846c/src/static/static/home/yi-hack-v5/script/cloudAPI_fake#L79

Its part of the busybox build in v5 so I will amend it soon

**Reply by alienatedsec** [reply]
> I presume that `cloudAPI` is executed by `cloud` - do we need to keep the cloud running regardless?

It does, that's how I found about the above `ntpd` error

```
/home/app # ./cloud
./cloudAPI_fake: line 79: /tmp/sd/yi-hack-v5/usr/sbin/ntpd: not found
^C
```

**Reply by roleoroleo** [reply]
Yes, cloudAPI is executed by cloud, so you don't need to run in system.sh.
About other yi processes:
- rmm, dispatch and cloud need to started even if DISABLE_CLOUD=yes
- mp4_record only if REC_WITHOUT_CLOUD=yes

**Reply by alienatedsec** [reply]
@roleoroleo how about
- ./p2p_tnp &
- ./oss &

?

**Reply by roleoroleo** [reply]
> @roleoroleo how about
> - ./p2p_tnp &
> - ./oss &
> 
> ?

Only if cloud is enabled.
Take a look at system.sh in MStar repo.

**Reply by kirilldem** [reply]
Could you share your camera config for frigate? I am getting a lot of errors on Frigate 12.1 Mine is like this:
```
   garden: # <------ Name the camera
     ffmpeg:
        hwaccel_args: preset-vaapi      
            # Optional: camera specific input args (default: inherit)
        input_args: -rtsp_transport tcp
        inputs:
         - path: rtsp://user:password@192.168.1.32/ch0_1.h264 # <----- low res
           roles:
             - detect
         - path: rtsp://user:password@192.168.1.32/ch0_0.h264
           roles:
             - record 
             - rtmp
```

**Reply by alienatedsec** [reply]
@Jakdaw in case you missed it
I think that @kirilldem is asking you to provide some support

**Reply by alienatedsec** [reply]
@kirilldem I recently have done some Frigate integration and described it more in detail [here](https://github.com/alienatedsec/yi-hack-v5/wiki/Integrating-with-Frigate)

**Reply by kirilldem** [reply]
Thanks for these settings, the errors seems to be gone in my log. 

I noticed one thing in your config - you set detect fps to 30, which is a lot of frames for detect, you should change it to around 5 fps to get better performance. See https://docs.frigate.video/frigate/camera_setup

**Reply by alienatedsec** [reply]
@kirilldem the stream has 30 fps, so the same needs to be on detect settings. The recommended value should be amended on the camera and in Frigate. As we cannot amend the FPS on the camera, you need to match the Frigate with the camera config.

**Comment by gaurishpathare** [comment]
Hi there.
I recently installed the update with latest version on my Yi Home camera. It works fine and connect to the app too but after my router restarted the webserver, rtsp feed and even ping stops working. Only the YI app is able to connect . Can you help suggest why this is happening. This happened once before and I had to re deploy the firmware hack to make it work again. 
@alienatedsec

**Reply by alienatedsec** [reply]
@gaurishpathare I guess once your router restarts, it gives other IP to your camera, which is why your YI app is still working.

**Comment by Helmuth1997** [comment]
![Screenshot_20240926-081925_YI Home](https://github.com/user-attachments/assets/f71f19d9-1456-411e-adb7-b0629a43a82a)
 
Alguna actualización para la Yi home 1080

**Reply by alienatedsec** [reply]
This hack does not support this camera firmware and chipset version 

Este truco no es compatible con esta versión de firmware y chipset de la cámara.

**Comment by liutas4x4** [comment]
- impossible to connect to ssh: no "root", nor "user" or "anonimous" hasn't access without password (as config is sayng);
- while router have been cofigured to static for this item, camera ingores it and goes to random, to DCHP.

**Comment by benatarmentia-ux** [comment]
Hi @alienatedsec,

I'm using a Yi Dome 1080p with official firmware 1.9.2.0I_201812141405.

I installed the official yi_dome_1080p_0.4.1.tgz successfully.

/home/yi-hack-v5 exists and reports version 0.4.1, and cloudAPI has been replaced correctly.

However, after every reboot, /tmp/sd is never mounted automatically, so system.sh is never executed and none of the yi-hack services (HTTP, RTSP, ONVIF, etc.) start.

If I manually mount the SD card:

mount -t vfat /dev/mmcblk0 /tmp/sd

and manually start BusyBox httpd, the WebUI works correctly.

Which component is responsible for mounting /tmp/sd and launching system.sh on firmware 1.9.2.0I_201812141405?

Is this a known issue or does it indicate that the installation did not complete correctly?

Thanks!
